mock/world/builtinFunctionsMock.go

```diff
@@ -83,7 +82,7 @@ func (b *MockWorld) runESDTTransferCall(vmInput *vmcommon.ContractCallInput) (*v
 		ReturnData:      make([][]byte, 0),
 		ReturnCode:      vmcommon.Ok,
 		ReturnMessage:   "",
-		GasRemaining:    vmInput.GasProvided - vmInput.GasLocked,
+		GasRemaining:    vmInput.GasProvided,
```

Comment:
Must deduct the gas used by the ESDT transfer, as specified by the current gas schedule.